### PR TITLE
Antlr4 ada202x

### PR DESCRIPTION
**Pull request by Lucretia**
Various fixes and update to draft 17.

### CODE REVIEW

File: antlr4/ada.g4 [hunk 1]
```diff
@@ -853,9 +872,49 @@ record_delta_aggregate :
 
 // 4.3.4
 array_delta_aggregate :
-   LPAREN expression WITH DELTA array_component_association_list RPAREN
+     LPAREN expression WITH DELTA array_component_association_list RPAREN
+   | LEFT_BRACKET expression WITH DELTA array_component_association_list RIGHT_BRACKET
+   ;
+
+// 4.3.5:
```
Comment (okellogg) [inline antlr4/ada.g4]: For uniformity with existing style, at RM section comments please omit trailing colon

File: antlr4/ada.g4 [hunk 2]
```diff
@@ -1275,6 +1364,53 @@ parameter_specification :
...
+// 6.1.2:
+global_set :
+     global_name ( COMMA global_name )*
+   | global_designator
+   ;
+
+// 6.1.2:
+global_designator : ALL | NuLL ;
+
+// TODO auxillary rule, (access_)subtype_mark
```
Comment (okellogg) [inline antlr4/ada.g4]:
```typo fix at "auxiliary"
// TODO auxiliary rule, (access_)subtype_mark
```

Comment (Lucretia) [reply antlr4/ada.g4]: ?

Comment (okellogg) [reply antlr4/ada.g4]: Typo "auxillary" -> _auxiliary_

**Comment by okellogg** [comment]
Thanks.

**Pull request merged**